ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SamlVersioningVerifier.kt

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.verification.core
+
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_2_3_3_a
+import org.codice.compliance.SAMLCore_3_2_2_c
+import org.codice.compliance.SAMLCore_4_1_2_a
+import org.codice.compliance.SAMLCore_4_1_3_2_a
+import org.codice.compliance.SAMLCore_4_1_3_2_b
+import org.codice.compliance.SAMLCore_4_1_3_3_a
+import org.codice.compliance.SAMLCore_4_2_a
+import org.codice.compliance.SAMLSpecRefMessage
+import org.codice.compliance.attributeNode
+import org.codice.compliance.recursiveChildren
+import org.codice.compliance.utils.TestCommon
+import org.opensaml.saml.common.SAMLVersion
+import org.w3c.dom.Node
+
+class SamlVersioningVerifier(private val samlResponseDom: Node) {
+    companion object {
+        private const val EXPECTED_MAJOR_VERSION = 2
+    }
+
+    /** 4 SAML Versioning */
+    fun verify() {
+        verifySetVersioning()
+        verifyAssertionVersioning()
+        verifyNamespaceVersioning()
+    }
+
+    /** 4.1 SAML Specification Set Version **/
+    @Suppress("SpreadOperator")
+    private fun verifySetVersioning() {
```

Comment:
❓ Are we only planning on calling this verification method during positive path testing?

If not, you are allowed to have mis-matched versions when returning an error response:
```
A SAML responder MUST NOT issue a response message with a major response version number lower than the major request version number of the corresponding request message except to report the error urn:oasis:names:tc:SAML:2.0:status:RequestVersionTooHigh
```